src/irmin-tezos/irmin_tezos.mli

```diff
@@ -26,6 +26,6 @@ module Store :
      and type Schema.Path.step = Schema.Path.step
      and type Schema.Contents.t = Schema.Contents.t
      and type Backend.Remote.endpoint = unit
-     and type contents_key = Schema.Hash.t Irmin_pack.Pack_key.t
-     and type node_key = Schema.Hash.t Irmin_pack.Pack_key.t
-     and type commit_key = Schema.Hash.t Irmin_pack.Pack_key.t
+     and type contents_key = Schema.Hash.t Irmin_pack_unix.Pack_key.t
```

Comment:
isn't it an issue to compile this codebase to javascript? (not sure exactly which part is required to compile to JS for Tezos)

Reply:
Hmm, that's a good question. I hadn't thought about JS compilation. Any pointers on where/how octez is compiled to JS? Also, is `irmin-tezos` used by octez?

Reply:
Thinking about this some more...

1. This constraint is a part of a `module Store = Irmin_pack_unix.S with ...` so it is already in the context of `irmin-pack.unix` (and therefore likely JS incompatible).
2. The current `Irmin_pack.Pack_key` is not implementation independent anyway -- it represents specifics of the unix implementation (with direct and indexed keys and inspection capabilities). This PR (hopefully) moves towards addressing this mixture.

So is this particular line an issue? I'm not sure but lean towards thinking it is not.

Also, here is some information I found for octez + JS

- https://tezos.gitlab.io/developer/javascript.html
- https://gitlab.com/tezos/tezos/-/blob/master/manifest/main.ml#L180-185

Reply:
`irmin-tezos/dune` already contains `irmin-pack.unix`. I think this is fine.